src/SAWScript/CrucibleBuiltins.hs

```diff
@@ -519,23 +522,29 @@ registerOverride ::
 registerOverride opts cc _ctx top_loc cs = do
   let sym = cc^.ccBackend
   sc <- CrucibleSAW.saw_ctx <$> liftIO (readIORef (W4.sbStateManager sym))
-  let fsym = (head cs)^.csName
+  let fstr = (head cs)^.csName
+      fsym = Text.pack fstr
       llvmctx = cc^.ccLLVMContext
+      matches (Crucible.LLVMHandleInfo _ h) =
+        Text.takeWhile (/= '.') (W4.functionName (Crucible.handleName h)) == fsym
   liftIO $
-    printOutLn opts Info $ "Registering override for `" ++ fsym ++ "`"
-  case Map.lookup (L.Symbol fsym) (llvmctx ^. Crucible.symbolMap) of
+    printOutLn opts Info $ "Registering override for `" ++ fstr ++ "`"
+  case filter matches (Map.elems (llvmctx ^. Crucible.symbolMap)) of
+    [] -> fail $ "Can't find declaration for `" ++ fstr ++ "`."
```

Comment:
If this user sees this message, will it be clear to them why SAW was looking for the declaration? Perhaps we could say `"Couldn't find declaration for fun while registering an override for it"`?

Reply:
Good point!